src/search/instamove/instamove.cc

```diff
@@ -121,43 +139,39 @@ class ValueHeadSearch : public InstamoveSearch {
         case GameResult::UNDECIDED:
           computation->AddInput(
               EvalPosition{history.GetPositions(), {}},
-              EvalResultPtr{.q = &results[i].q, .d = &results[i].d});
+              EvalResultPtr{.q = &results[i].negative_q, .d = &results[i].d});
           break;
         case GameResult::DRAW:
-          results[i].q = 0;
-          results[i].d = 1;
+          results[i] = {.negative_q = 0, .d = 1, .mate = std::nullopt};
           break;
         default:
           // A legal move to a non-drawn terminal without tablebases must be a
           // win.
-          results[i].q = -1;
-          results[i].d = 0;
+          results[i] = {.negative_q = -1, .d = 0, .mate = 1};
       }
       history.Pop();
     }
 
     computation->ComputeBlocking();
 
     const size_t best_idx =
-        std::min_element(results.begin(), results.end(),
-                         [](const EvalResult& a, const EvalResult& b) {
-                           return a.q < b.q;
-                         }) -
-        results.begin();
-
-    std::vector<ThinkingInfo> infos = {{
-        .depth = 1,
-        .seldepth = 1,
-        .nodes = static_cast<int64_t>(legal_moves.size()),
-        .score = 90 * std::tan(1.5637541897 * results[best_idx].q),
-        .wdl =
-            ThinkingInfo::WDL{
-                static_cast<int>(std::round(
-                    500 * (1 + results[best_idx].q - results[best_idx].d))),
-                static_cast<int>(std::round(1000 * results[best_idx].d)),
-                static_cast<int>(std::round(
-                    500 * (1 - results[best_idx].q - results[best_idx].d)))},
-    }};
+        std::min_element(results.begin(), results.end()) - results.begin();
+
+    const Score& r = results[best_idx];
+    auto to_int = [](double x) { return static_cast<int>(std::round(x)); };
+    std::vector<ThinkingInfo> infos{
+        {.depth = 1,
+         .seldepth = 1,
+         .nodes = static_cast<int64_t>(legal_moves.size()),
+         .mate = r.mate,
+         .score = -90 * std::tan(1.5637541897 * r.negative_q),
+         .wdl = r.mate
```

Comment:
I just checked and normal search reports wdl with mate score, so maybe this isn't needed.